src/Symfony/Component/PasswordHasher/Hasher/UserPasswordHasher.php

```diff
@@ -31,10 +31,7 @@ public function __construct(PasswordHasherFactoryInterface $hasherFactory)
         $this->hasherFactory = $hasherFactory;
     }
 
-    /**
-     * @param PasswordAuthenticatedUserInterface $user
-     */
-    public function hashPassword($user, string $plainPassword): string
+    public function hashPassword(UserInterface|string|PasswordAuthenticatedUserInterface $user, string $plainPassword): string
```

Comment:
strings are not supported in UserPasswordHasher.

And anything else than `PasswordAuthenticatedUserInterface` triggers a deprecation (and so will be removed before the 6.0 release), so the typehint should be `PasswordAuthenticatedUserInterface`, not `PasswordAuthenticatedUserInterface|UserInterface`

Reply:
same in other methods

Reply:
I tried but the function will need some rework if we put only  `PasswordAuthenticatedUserInterface` , there is an error for `$user->getSalt` but I don't know how to handle it

Reply:
Well, that will be solved when cleaning deprecated APIs. In 6.0, `getSalt` should be called only if the User implements `LegacyPasswordAuthenticatedUserInterface`.

The issue here is that you try to implement the new typehints _before_ cleaning the deprecated code, and so you still need to account for old types while 6.0 won't need that.

Reply:
But what appends if the user is not `LegacyPasswordAuthenticatedUserInterface` ?

Reply:
`$salt = null` in Symfony 6

Reply:
Ok thank you!